src/artifacts_poetry_plugin/plugin.py

```diff
@@ -4,23 +4,139 @@
 import poetry.console.application
 from poetry.factory import Factory
 from cleo.commands.command import Command
+import poetry.config.config
+from poetry.publishing.uploader import Uploader
+from poetry.utils.authenticator import Authenticator
+from cleo.helpers import argument
+import logging
 
-ARTIFACTS_LIST = "artifacts-list"
+ARTIFACTS_DEPLOY = "artifacts-deploy"
+REPOSITORY_NAME = "repository_name"
+logger = logging.getLogger(__name__)
 
 
-def getPackages(poetry):
+def get_repository_data(config, repo_name, io):
+    """Gets the URL, username, and password from the poetry configuration for the given repo name"""
+    repo_data = {}
+    repo_data["url"] = config.get(f"repositories.{repo_name}.url")
+    if repo_data["url"] is None:
```

Comment:
Raising an exception this way is fine, but you can also add a parameter to the .get() call that will do the same thing.  i.e.  config.get('somekey', exec("raise RuntimeError(f'Repository {repo_name} is not defined') ")).  Just for reference

Reply:
I left it the way it was since the field is accessed later.